### DISCUSSION TITLE
Can I use initializeApp() for supporting different environments ?

### DISCUSSION THREAD

**Discussion by elirangoshen** [invertase/react-native-firebase#6418] [Q&A]
I want to support dev staging and prod environments with corresponding firebase projects. 
I would like to use only code changes rather than file changes to make it easier to manage and i notice that I can use initializeApp function for switching projects from the code..

Does someone used it for this purpose ? should it work ?

**Comment by mikehardy** [comment]
This won't work with only code changes, because the Google services plist/json files define the "DEFAULT" app and that is not something you can change for some of the modules. There are discussions of this in the issue tracker and maybe other issues - I personally use android build flavors and ios build targets to have different files included (depending on the flavor or target) but that implies different package names / bundle ids so I have fully separate apps, one app per backend.

I personally like this though, as that means I can have each app (with separate backend) installed at the same time on a phone at the same time
